subprojects/core/src/main/java/org/gradle/internal/classpath/transforms/InstrumentingClassTransform.java

```diff
@@ -497,15 +284,51 @@ private static String getBoostrapMethodName(BytecodeInterceptorFilter intercepto
             }
         }
 
-        private String binaryClassNameOf(String className) {
-            return getObjectType(className).getClassName();
-        }
-
         private boolean isGroovyIndyCallsite(Handle bootstrapMethodHandle) {
             return (bootstrapMethodHandle.getOwner().equals(GROOVY_INDY_INTERFACE_TYPE) ||
                 bootstrapMethodHandle.getOwner().equals(GROOVY_INDY_INTERFACE_V7_TYPE)) &&
                 bootstrapMethodHandle.getName().equals("bootstrap") &&
                 bootstrapMethodHandle.getDesc().equals(GROOVY_INDY_INTERFACE_BOOTSTRAP_METHOD_DESCRIPTOR);
         }
+
+        private Object maybeInstrumentBootstrapArgument(Object argument) {
+            if (argument instanceof Handle) {
+                return maybeInstrumentHandle((Handle) argument);
+            }
+            return argument;
+        }
+
+        private Handle maybeInstrumentHandle(Handle handle) {
+            BridgeMethod bridgeMethod = owner.bridgeMethods.computeIfAbsent(handle, this::maybeBuildBridgeMethod);
+            if (bridgeMethod != null) {
+                return bridgeMethod.bridgeMethodHandle;
+            }
+            return handle; // No instrumentation requested.
+        }
+
+        @Nullable
+        private BridgeMethod maybeBuildBridgeMethod(Handle handle) {
+            for (JvmBytecodeCallInterceptor interceptor : interceptors) {
+                BridgeMethodBuilder methodBuilder = interceptor.findBridgeMethodBuilder(className, handle.getTag(), handle.getOwner(), handle.getName(), handle.getDesc());
+                if (methodBuilder != null) {
+                    return new BridgeMethod(makeBridgeMethodHandle(makeBridgeMethodName(handle), methodBuilder.getBridgeMethodDescriptor()), methodBuilder);
+                }
+            }
+            return null;
+        }
+
+        private String makeBridgeMethodName(Handle originalHandle) {
+            int index = owner.bridgeMethods.size();
+            // com/foo/Bar -> com$foo$Bar
+            String mangledOwner = originalHandle.getOwner().replace("/", "$");
+            // Only <init> and <clinit> are allowed to have <> in the name.
+            // As we're intercepting these too, we strip prohibited symbols from the bridge method's name.
+            String safeName = originalHandle.getName().replace("<", "").replace(">", "");
```

Comment:
❓ Is it possible that methods with name `init` conflict with this replacement? 
Should we replace `<` with something else than empty string?

Reply:
>Is it possible that methods with name init conflict with this replacement?
 
Not really, because each generated method gets a unique index. So there will be `foo$$init$0` and `foo$$init$1`. However, there is a potential to confuse them in the stack trace, so I changed `<init>` to be `_init_` rather than just `init`.